fe/fe-core/src/main/java/com/starrocks/connector/elasticsearch/ElasticsearchConnector.java

```diff
@@ -0,0 +1,135 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package com.starrocks.connector.elasticsearch;
+
+import com.starrocks.catalog.EsTable;
+import com.starrocks.common.DdlException;
+import com.starrocks.connector.Connector;
+import com.starrocks.connector.ConnectorContext;
+import com.starrocks.connector.ConnectorMetadata;
+import com.starrocks.connector.elasticsearch.external.EsRestClient;
+import com.starrocks.connector.elasticsearch.external.EsUtil;
+import com.starrocks.connector.exception.StarRocksConnectorException;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Map;
+
+public class ElasticsearchConnector
+        implements Connector {
+    private static final Logger LOG = LogManager.getLogger(ElasticsearchConnector.class);
+
+    private EsRestClient esRestClient;
+
+    private final String catalogName;
+
+    private String[] nodes;
+
+    private String username = null;
+
+    private String password = null;
+
+    private boolean enableDocValueScan = true;
+
+    private boolean enableKeywordSniff = true;
+
+    private boolean enableSsl = false;
+
+    private boolean enableWanOnly = true;
+
+    private Map<String, String> properties;
+
+    private ConnectorMetadata metadata;
+
+    /**
+     * Default constructor for EsExternalCatalog.
+     */
+    public ElasticsearchConnector(ConnectorContext contex) throws DdlException {
+        this.catalogName = contex.getCatalogName();
+        this.properties = processDefaultValue(contex.getProperties());
+        this.esRestClient = new EsRestClient(this.nodes, this.username, this.password, this.enableSsl);
+    }
+
+    /**
+     * add some default value
+     */
+    private Map<String, String> processDefaultValue(Map<String, String> properties) throws DdlException {
+        nodes = properties.get(EsTable.HOSTS).trim().split(",");
```

Comment:
The class field `enableDocValueScan`, `enableKeywordSniff`,`enableWanOnly` are not used. and I think the `nodes`/`username`/`password`/`enableSsl` aren't necessarily placed in this class. They seem just to be building esClient for now.